lib/compiler.js

```diff
@@ -402,6 +402,14 @@ function buildMiddlewareInstructions(rootDir, config) {
       allConfigs.forEach(function(config) {
         var resolved = resolveMiddlewarePath(rootDir, middleware);
 
+        // prioritize coffee scripts over json
+        if (!isPreferredExtension(resolved.sourceFile)) {
+          var otherFile = tryOtherExtensions(resolved.sourceFile);
+          if (otherFile !== undefined) {
+            resolved.sourceFile = otherFile;
+          }
```

Comment:
What would be even better: a function called e.g. `resolveAppScriptPath` that would call `resolveAppPath` under the hood and then `fixFileExtension` too. That should reduce even more the amount of changes needed.

Since `resolveMiddlewarePath` is calling `resolveAppPath` under the hood, you can call the new method there too and that way the callers of `resolveMiddlewarePath` don't need to know about these implementation details at all.